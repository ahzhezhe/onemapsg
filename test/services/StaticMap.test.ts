import { oneMap } from '..';

test('getStaticImageUri', () => {
  const result = oneMap.staticMap.getStaticImageUri({
    layerchosen: 'night',
    lat: 1.3039454,
    lng: 103.8408471,
    postal: '542226',
    zoom: 15,
    width: 500,
    height: 500,
    polygons: [
      {
        points: [[1.3039454, 103.8408471], [1.3039454, 103.8408471]],
        color: [50, 100, 150]
      },
      {
        points: [[1.3039454, 103.8408471], [1.3039454, 103.8408471]],
        color: [50, 100, 150]
      }
    ],
    lines: [
      {
        points: [[1.3039454, 103.8408471], [1.3039454, 103.8408471]],
        color: [50, 100, 150],
        thickness: 3
      },
      {
        points: [[1.3039454, 103.8408471], [1.3039454, 103.8408471]],
        color: [50, 100, 150],
        thickness: 3
      }
    ],
    points: [
      {
        point: [1.3039454, 103.8408471],
        color: [50, 100, 150],
        label: 'A'
      },
      {
        point: [1.3039454, 103.8408471],
        color: [50, 100, 150],
        label: 'B'
      }
    ],
    color: [50, 100, 150],
    fillColor: [50, 100, 150]
  });
  expect(result).toMatchSnapshot();
});
