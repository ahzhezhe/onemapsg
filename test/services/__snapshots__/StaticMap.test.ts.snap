// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getStaticImageUri 1`] = `"https://developers.onemap.sg/commonapi/staticmap/getStaticImage?layerchosen=night&lat=1.3039454&lng=103.8408471&postal=542226&zoom=15&width=500&height=500&polygons=%5B%5B1.3039454%2C103.8408471%5D%2C%5B1.3039454%2C103.8408471%5D%5D%3A50%2C100%2C150%7C%5B%5B1.3039454%2C103.8408471%5D%2C%5B1.3039454%2C103.8408471%5D%5D%3A50%2C100%2C150&lines=%5B%5B1.3039454%2C103.8408471%5D%2C%5B1.3039454%2C103.8408471%5D%5D%3A50%2C100%2C150%3A3%7C%5B%5B1.3039454%2C103.8408471%5D%2C%5B1.3039454%2C103.8408471%5D%5D%3A50%2C100%2C150%3A3&points=%5B1.3039454%2C103.8408471%2C%2250%2C100%2C150%22%2C%22A%22%5D%7C%5B1.3039454%2C103.8408471%2C%2250%2C100%2C150%22%2C%22B%22%5D&color=50%2C100%2C150&fillColor=50%2C100%2C150"`;
